0xPolygon/polygon-edge
# Add custom marshalling and unmarshalling for block Header

# Description

Polygon Edge Header cannot be unmarshalled with go-etherem ethClient. This PR implements custom json marshal and unrmashal for Header so it is compatible with other EVM networks

# Changes include


# Breaking changes

I don't think this is a breaking change
I've manually tested:
started a network on the latest develop branch, switching to this branch, adding a new node to bulkSync, waiting in watchSync, voting to become validator, watching consensus 

# Checklist


## Testing


### Manual tests

I've created go-ethereum ethClient script to test this feature. If you run this script on develop branch you will see bunch of errors
```

package main

import (
	"context"
	"fmt"
	"log"

	"github.com/ethereum/go-ethereum/core/types"
	"github.com/ethereum/go-ethereum/ethclient"
)

func main() {
	client, err := ethclient.Dial("ws://localhost:10002/ws")
	if err != nil {
		log.Fatal(err)
	}

	headers := make(chan *types.Header)
	sub, err := client.SubscribeNewHead(context.Background(), headers)
	if err != nil {
		log.Fatal(err)
	}

	for {
		select {
		case err := <-sub.Err():
			log.Fatal(err)
		case header := <-headers:
			fmt.Println(header)

		}
	}
}
```

# Additional comments

Fixes EDGE-440
